code/modules/ruins/lavalandruin_code/techcult.dm

```diff
@@ -0,0 +1,592 @@
+	dose_available = FALSE
+	addtimer(VARSET_CALLBACK(src, dose_available, TRUE), 2 MINUTES)
+
+// Reviver Implant PLUS
+/obj/item/organ/cyberimp/chest/reviver/plus
+	name = "Reviver implant PLUS"
+	desc = "This implant will attempt to heal you REALLY FAST if you lose consciousness. For the true warriors!"
+	implant_color = "#CC0605" //Cooler red
+
+/obj/item/organ/cyberimp/chest/reviver/plus/on_life()
+	if(reviving)
+		if(owner.stat == UNCONSCIOUS)
+			addtimer(CALLBACK(src, .proc/heal), 10) //Fast
+		else
+			cooldown = revive_cost + world.time
+			reviving = FALSE
+			to_chat(owner, "<span class='notice'>Your reviver implant shuts down and starts recharging. It will be ready again in [DisplayTimeText(revive_cost)].</span>")
+		return
+
+	if(cooldown > world.time)
+		return
+	if(owner.stat != UNCONSCIOUS)
+		return
+	if(owner.suiciding)
+		return
+
+	revive_cost = 0
+	reviving = TRUE
+	to_chat(owner, "<span class='notice'>You feel a faint buzzing as your reviver implant starts patching your wounds...</span>")
+
+/obj/item/organ/cyberimp/chest/reviver/plus/heal()
+	if(owner.getOxyLoss())
+		owner.adjustOxyLoss(-10) //REAL fast
+		revive_cost += 5
+	if(owner.getBruteLoss())
+		owner.adjustBruteLoss(-5)
+		revive_cost += 40
+	if(owner.getFireLoss())
+		owner.adjustFireLoss(-5)
+		revive_cost += 40
+	if(owner.getToxLoss())
+		owner.adjustToxLoss(-3)
+		revive_cost += 40
+
+//Advanced Surgical Toolset Implant
+/obj/item/organ/cyberimp/arm/surgery/plus
+	name = "advanced surgical toolset implant"
+	desc = "A set of advanced surgical tools hidden behind a concealed panel on the user's arm."
+	contents = newlist(/obj/item/retractor/advanced/augment, /obj/item/surgicaldrill/advanced/augment, /obj/item/scalpel/advanced/augment, /obj/item/surgical_drapes)
+
+/obj/item/organ/cyberimp/arm/surgery/plus/emag_act(mob/user)
+	if(!(locate(/obj/item/reagent_containers/borghypo/hacked/augment) in items_list))
+		to_chat(user, "<span class='notice'>You unlock and hack integrated hypospray located in [src]!</span>") //Oh god oh fuck
+		items_list += new /obj/item/reagent_containers/borghypo/hacked/augment(src)
+		return 1
+	return 0
+
+/obj/item/retractor/advanced/augment
+	toolspeed = 0.3 //I AM SPEED!!
+
+/obj/item/surgicaldrill/advanced/augment
+	toolspeed = 0.3 //Still not as fast as alien tools though.
+
+/obj/item/scalpel/advanced/augment
+	toolspeed = 0.3
+
+/obj/item/reagent_containers/borghypo/hacked/augment //Kill people with toxins!
+	name = "hypospray"
+	desc = "A very dangerous tool that is able to synthesize even more dangerous chemicals."
+	recharge_time = 30 //10u every minute
+
+/obj/item/reagent_containers/borghypo/hacked/augment/regenerate_reagents()
+	for(var/i in 1 to reagent_ids.len)
+		var/datum/reagents/RG = reagent_list[i]
+		if(RG.total_volume < RG.maximum_volume)
+			RG.add_reagent(reagent_ids[i], 5)
+
+/obj/item/reagent_containers/borghypo/hacked/augment/add_reagent(datum/reagent/reagent)
```

Comment:
check to see if the parent item already has this stuff

Reply:
Parent has this stuff, but it's a shit-code.
Volume variable is never used so I'm calling this to change it from 30 to 10.